outputs/jira_test.go

```diff
@@ -0,0 +1,137 @@
+package outputs
+
+import (
+	"github.com/aquasecurity/go-jira"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"testing"
+)
+
+func TestСreateIssuePriority(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	tests := []struct {
+		name         string
+		jiraAPI      *JiraAPI
+		priorities   interface{}
+		wantPriority string
+		wantError    string
+	}{
+		{
+			name:         "happy path (empty priority, jira have 'High' field)",
+			jiraAPI:      &JiraAPI{Priority: ""},
+			priorities:   []jira.Priority{{Name: "Highest"}, {Name: "High"}, {Name: "Medium"}},
+			wantPriority: "High",
+		},
+		{
+			name:         "happy path (empty priority, jira doesn't have 'High' field)",
+			jiraAPI:      &JiraAPI{Priority: ""},
+			priorities:   []jira.Priority{{Name: "Highest"}, {Name: "Low"}, {Name: "Medium"}},
+			wantPriority: "Highest",
+		},
+		{
+			name:         "happy path (fill priority, jira have 'Medium' field)",
+			jiraAPI:      &JiraAPI{Priority: "Medium"},
+			priorities:   []jira.Priority{{Name: "Highest"}, {Name: "High"}, {Name: "Medium"}},
+			wantPriority: "Medium",
+		},
+		{
+			name:       "bad path (fill priority, jira doesn't have 'Medium' field)",
+			jiraAPI:    &JiraAPI{Priority: "Medium"},
+			priorities: []jira.Priority{{Name: "Highest"}, {Name: "High"}, {Name: "Low"}},
+			wantError:  "project don't have issue priority \"Medium\"",
+		},
+		{
+			name:       "bad path (jira returns empty priorities)",
+			jiraAPI:    &JiraAPI{Priority: ""},
+			priorities: nil,
+			wantError:  "project don't have issue priorities",
+		},
+		{
+			name:       "bad path (jira returns error)",
+			jiraAPI:    &JiraAPI{Priority: ""},
+			priorities: jira.Issue{},
+			wantError:  "failed to get issue priority list: json: cannot unmarshal object into Go value of type []jira.Priority",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			httpmock.RegisterResponder("GET", "http://testUrl/rest/api/2/priority",
+				func(req *http.Request) (*http.Response, error) {
+					resp, err := httpmock.NewJsonResponse(200, test.priorities)
+					return resp, err
+				},
+			)
```

Comment:
Any reason for using this dependency? Standard library should work fine as so:


```
 38  	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    39  		fmt.Fprintln(w, test.priorities)
    40  	}))
    41  	defer ts.Close()
```

Then use `ts.URL()` as the URL.

Best to not use external dependencies if standard library would meet the need.

Reply:
Okay, thanks. I changed the dependency to `httptest`.